src/ConcordiumNetSdk/Types/ContractAddress.cs

```diff
@@ -1,62 +1,88 @@
-using System.Buffers.Binary;
+using ConcordiumNetSdk.Helpers;
 
 namespace ConcordiumNetSdk.Types;
 
 /// <summary>
-/// Represents the information about a contract address.
+/// Represents a contract address.
+///
+/// A contract is identified by its unique <see cref="ContractAddress"/>.
+/// A contract address consists of a <see cref="ContractIndex"/> and a
+/// <see cref="ContractSubIndex"/> both represented as 64-bit unsigned
+/// integers.
 /// </summary>
-public class ContractAddress
+public readonly struct ContractAddress : IEquatable<ContractAddress>
 {
+    public const UInt32 BytesLength = ContractIndex.BytesLength + ContractSubIndex.BytesLength;
+
     /// <summary>
-    /// Gets the serialized bytes length.
+    /// The index part of the address of a contract.
     /// </summary>
-    public const int BytesLength = 16;
+    public readonly ContractIndex _index { get; init; }
```

Comment:
I think having these ContractIndex/Subindex types is too heavy for the SDK. They don't really add any type safety and just add complexity. I recommend removing them.